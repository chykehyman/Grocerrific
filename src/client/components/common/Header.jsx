import React from 'react';
import { NavLink } from 'react-router-dom';

import logo from '../../assets/images/app-logo.png';


export const Header = () => (
  <div className="container header-container">
    <NavLink to="/">
      <img src={logo} alt="site_logo" width="55" height="55" />
    </NavLink>
    <nav>
      <NavLink exact to="/" className="nav-link" activeClassName="active">Home</NavLink>
      <NavLink exact to="/shop" className="nav-link" activeClassName="active">Shop</NavLink>
    </nav>
  </div>
);

export default Header;
