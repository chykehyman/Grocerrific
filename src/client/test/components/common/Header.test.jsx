import Header from '../../../components/common/Header';

describe('<Header /> component', () => {
  const shallowWrapper = shallow(<Header />);

  it('renders the header component without crashing', () => {
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should have two children', () => {
    expect(shallowWrapper.find('.header-container').props().children).toHaveLength(2);
    expect(shallowWrapper.find('NavLink').first().prop('to')).toBe('/');
    expect(shallowWrapper.find('NavLink').last().prop('to')).toBe('/shop');
  });

  describe('Proper NavLink switching', () => {
    const renderHeader = path => mount(
      <MemoryRouter initialEntries={[path]}>
        <Header />
      </MemoryRouter>
    );
    const expectActiveLink = component => expect(
      component
        .find('.active').at(2).text()
    );
    it('is active when we are in the home route', () => {
      const component = renderHeader('/');

      expectActiveLink(component).toBe('Home');
    });
    it('is active when we switch to shop route', () => {
      const component = renderHeader('/shop');

      expectActiveLink(component).toBe('Shop');
    });
  });
});
