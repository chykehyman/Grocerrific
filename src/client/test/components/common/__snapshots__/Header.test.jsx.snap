// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component renders the header component without crashing 1`] = `
<div
  className="container header-container"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  >
    <img
      alt="site_logo"
      height="55"
      src={
        Object {
          "process": [Function],
        }
      }
      width="55"
    />
  </NavLink>
  <nav>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-link"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-link"
      exact={true}
      to="/shop"
    >
      Shop
    </NavLink>
  </nav>
</div>
`;
